plugin/db/mysql/dump.go

```diff
@@ -226,26 +234,67 @@ func dumpTxn(ctx context.Context, txn *sql.Tx, database string, out io.Writer, s
 			}
 		}
 
+		// Disable foreign key check.
+		// mysqldump uses the same mechanism. When there is any schema or data dependency, we have to disable
+		// the unique and foreign key check so that the restoring will not fail.
+		if _, err := io.WriteString(out, disableUniqueAndForeignKeyCheckStmt); err != nil {
+			return err
+		}
+
 		// Table and view statement.
+		// We have to dump the table before views because of the structure dependency.
 		tables, err := getTablesTx(txn, dbName)
 		if err != nil {
 			return errors.Wrapf(err, "failed to get tables of database %q", dbName)
 		}
+		// Construct temporal views.
+		// Create a table with the same name as the view and with columns of
+		// the same name in order to satisfy views that depend on this view.
+		// The table will be removed when the actual view is created.
+		// The properties of each column, are not preserved in this temporary
+		// table, because they are not necessary.
+		// This will not be necessary once we can determine dependencies
+		// between views and can simply dump them in the appropriate order.
+		// https://sourcegraph.com/github.com/mysql/mysql-server/-/blob/client/mysqldump.cc?L2781
+		for _, tbl := range tables {
+			if tbl.TableType != viewTableType {
+				continue
+			}
+			if _, err := io.WriteString(out, fmt.Sprintf("%s\n", getTemporalView(tbl.Name, tbl.ViewColumns))); err != nil {
+				return err
+			}
+		}
+		// Construct tables.
 		for _, tbl := range tables {
-			if schemaOnly && tbl.TableType == baseTableType {
+			if tbl.TableType != baseTableType {
+				continue
+			}
+			if schemaOnly {
 				tbl.Statement = excludeSchemaAutoIncrementValue(tbl.Statement)
 			}
 			if _, err := io.WriteString(out, fmt.Sprintf("%s\n", tbl.Statement)); err != nil {
 				return err
 			}
-			if !schemaOnly && tbl.TableType == baseTableType {
+			if !schemaOnly {
 				// Include db prefix if dumping multiple databases.
 				includeDbPrefix := len(dumpableDbNames) > 1
 				if err := exportTableData(txn, dbName, tbl.Name, includeDbPrefix, out); err != nil {
 					return err
 				}
 			}
 		}
+		// Construct final views.
+		for _, tbl := range tables {
+			if tbl.TableType != viewTableType {
+				continue
+			}
+			if _, err := io.WriteString(out, fmt.Sprintf("DROP VIEW IF EXISTS `%s`;\n", tbl.Name)); err != nil {
```

Comment:
add a comment to explain we first remove the just created temporary view

Reply:
Done.